cpp/simulations/abm.cpp

```diff
@@ -864,20 +866,84 @@ int main()
-        fprintf(f_abm, "%f ", sim.get_result().get_time(i));
-        auto v = sim.get_result().get_value(i);
+    for (auto i = 0; i < results.get_num_time_points(); ++i) {
+        fprintf(f_abm, "%f ", results.get_time(i));
+        auto v = results.get_value(i);
         for (auto j = 0; j < v.size(); ++j) {
             fprintf(f_abm, "%f", v[j]);
             if (j < v.size() - 1) {
                 fprintf(f_abm, " ");
             }
         }
-        if (i < sim.get_result().get_num_time_points() - 1) {
+        if (i < results.get_num_time_points() - 1) {
             fprintf(f_abm, "\n");
         }
     }
     fclose(f_abm);
 }
+
+/**
+ * Run the abm simulation
+ * @param result_dir directory where all results of the parameter study will be stored.
+ * @param save_single_runs [Default: true] Defines if single run results are written to the disk.
+ * @returns any io error that occurs during reading or writing of files.
+ */
+mio::IOResult<void> run(const fs::path& result_dir, bool save_single_runs = true)
+{
+
+    auto t0         = mio::abm::TimePoint(0); // Start time (t0) per simulation
+    auto tmax       = mio::abm::TimePoint(0) + mio::abm::days(60); // Endtime (tmax) per simulation
+    const auto num_runs     = 1; // Number of runs
+    auto ensemble_results = std::vector<mio::TimeSeries<double>>{}; // Vector of collected results (TimeSeries objects)
+    ensemble_results.reserve(size_t(num_runs));
+    auto run_idx            = size_t(1); //The run index
+    auto save_result_result = mio::IOResult<void>(mio::success());
+
+    // Loop over a number of run
+    for (size_t i = 0; i < num_runs; i++) {
+        // Create the sampled simulation with start time t0
+        auto sim = create_sampled_simulation(t0);
+        // Advance the world to tmax
+        sim.advance(tmax);
+        // Push result of the simulation back to the result vector
+        ensemble_results.push_back(sim.get_result());
+        // Option to save the current run result to file
+        if (save_result_result && save_single_runs) {
+            save_result(sim.get_result(), run_idx, result_dir);
+        }
+        ++run_idx;
+    }
+    BOOST_OUTCOME_TRY(save_result_result);
+    return mio::success();
+}
+
+int main(int argc, char** argv)
+{
+
+    mio::set_log_level(mio::LogLevel::warn);
+
+    std::string result_dir = ".";
+    bool save_single_runs = true;
+
+    if (argc == 2) {
+        result_dir       = argv[1];
+    }
+    printf("Saving results to \"%s\".\n", result_dir.c_str());
+
+    // mio::thread_local_rng().seed({...}); //set seeds, e.g., for debugging
+    //printf("Seeds: ");
+    //for (auto s : mio::thread_local_rng().get_seeds()) {
+    //    printf("%u, ", s);
+    //}
+    //printf("\n");
+
+    auto result = run(result_dir, save_single_runs);
+    if (!result) {
+        printf("%s\n", result.error().formatted_message().c_str());
+        return -1;
+    }
+    return 0;
+}
```

Comment:
According to some prior issues, we should always end a file with a newline. Please check for all files.

Reply:
It's done in the latest update.